### DISCUSSION TITLE
Can't find dependent libraries in Windows

### DISCUSSION THREAD

**Discussion by yxmm-wxe** [duckdb/duckdb#2601] [Q&A] [answered]
When I use DuckDB via duckdb_jdbc 0.3.0 in Windows, for some machines, it will throw the following issue.

```
java.lang.UnsatisfiedLinkError: C:\Windows\Temp\libduckdb_java1570861608042760038.so: Can't find dependent libraries
``` 

I guess it is the problem of permission. But I'm not sure.

Is 'libduckdb_java1570861608042760038.so' generated by duckdb_jdbc?

Furthermore, I'd like to know whether the directory of 'libduckdb_java1570861608042760038.so' can be specified.

**Comment by hannes** [comment]
Can you be more specific please? "Some" machines, which operating system etc? My guess would be the visual C runtime is not installed, but hard to tell.

**Reply by yxmm-wxe** [reply]
Sorry for my fuzzy description. I meant that in some virtual machines with Windows Enterprise installed, the issue can be reproduced.

On the virtual machine, only the following are installed

- Microsoft .NET Core 3.1.7 - Windows Server Hosting
- Microsoft .NET Core Runtime - 3.1.7 (x64)
- Microsoft .NET Core Runtime - 3.1.7 (x86)
- Microsoft Visual C++ 2013 Redistributable (x64) - 12.0.30501
- Microsoft Visual C++ 2015 Redistributable (x64) - 14.0.23918

I don't find anything about visual C runtime. Could you specify the visual C runtime? 

Otherwise, 'C:\Windows\Temp\libduckdb_java1570861608042760038.so' does exist.

**Comment by hannes** [comment, marked as answer]
The file is generated at runtime, its temporary. Try installing the visual c++ runtime redistributable please, see here https://docs.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170

**Reply by yxmm-wxe** [reply]
Thanks a lot! It works now.

**Reply by 3dot141** [reply]
Could i static link this to avoid it ?
